service/src/main/java/crawlercommons/urlfrontier/service/memory/InternalURL.java

```diff
@@ -15,7 +15,9 @@
  * simpler than the objects from gRPC + sortable and have equals based on URL only. The metadata key
  * values are compressed into a single byte array.
  */
-class InternalURL implements Comparable<InternalURL>, Serializable {
+public class InternalURL implements Comparable<InternalURL>, Serializable {
```

Comment:
This class is not meant to be used outside the package. Any reason you needed it to be public?

Reply:
Indeed, this is not needed. It was a leftover from a previous interim release where I was returning InternalURL instead of URLItem